### PR TITLE
Mail listner bug

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47530

## Description
Skipped emails with no date field.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def __init__(self, message_bytes: bytes, include_raw_body: bool, save_file: bool
 
             message_bytes = self.handle_message_slashes(message_bytes)
             email_object = parse_from_bytes(message_bytes)
-
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:33]: restore

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 2]
```diff
@@ -231,6 +231,7 @@ def fetch_incidents(client: IMAPClient,
         next_run: This will be last_run in the next fetch-incidents
         incidents: Incidents that will be created in Demisto
     """
+    LOG('in fetch incidents function')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:234]: replace with a `logger` decorator

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 3]
```diff
@@ -474,7 +479,9 @@ def main():
         ssl_context.verify_mode = ssl.CERT_NONE
     LOG(f'Command being called is {demisto.command()}')
     try:
+        LOG('before creating client')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:482]: remove

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 4]
```diff
@@ -474,7 +479,9 @@ def main():
...
         with IMAPClient(mail_server_url, ssl=tls_connection, port=port, ssl_context=ssl_context) as client:
+            LOG('after creating client')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:484]: remove

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 5]
```diff
@@ -369,6 +373,7 @@ def generate_search_query(time_to_fetch_from: Optional[datetime],
     Returns:
         A list with arguments for the email search query
     """
+    LOG('in generate search query')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:376]: Replace with the `logger` decorator

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 6]
```diff
@@ -231,7 +232,7 @@ def fetch_incidents(client: IMAPClient,
         next_run: This will be last_run in the next fetch-incidents
         incidents: Incidents that will be created in Demisto
     """
-    LOG('in fetch incidents function')
+    demisto.debug('in fetch incidents function')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:235]: use the [logger function](https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L1679) instead

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 7]
```diff
@@ -373,7 +374,7 @@ def generate_search_query(time_to_fetch_from: Optional[datetime],
     Returns:
         A list with arguments for the email search query
     """
-    LOG('in generate search query')
+    demisto.debug('in generate search query')
```
Comment (yaakovi) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:377]: use the [logger function](https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L1679) instead

**Review by yaakovi — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Pull request merged**